### ISSUE TITLE
Document @com_google_protobuf//:protoc?

### ISSUE THREAD

**Issue by liningpan** [bazelbuild/rules_proto#174]
There are some rust protobuf/grpc libraries (e.g. prost, tonic) that use the `protoc` binary to compile `.proto` files into the descriptor set. Afterwards, these libraries use the descriptor set to generate rust code. 

I found out that `@com_google_protobuf//:protoc` is a convenient way to get precompiled `protoc` (whenever possible). However, I can't find where this is documented or if this is something stable that we can rely on.

**Comment by comius** [comment]
Correct, `@com_google_protobuf//:protoc` handles precompiled `protoc`.

It's quite hacky, as it constructs a pre-bzlmod repository, which overwrites protoc published BUILD files with targets that select binaries. Implementation is in rules_proto and it's hard to follow.

The problem with it is, that the versions need to be updated on each protoc release in rules_proto repository (hence the 21.7 in the name of protoc release).

To fully support bzlmod we need another mechanism. This mechanism are toolchains, which make it possible to select binary if one is available on the platform (cpu/operating system) and fallback to sources if not.

When toolchains are implemented the hack will be removed from this repository.
